### PR TITLE
[scripts/generate_dump] add information to tech-support file

### PR DESCRIPTION
**Pull request by LimorAmit**
#### What I did
I added/extended some commands of the generate_dump script to collect more information about the system and processes.
In addition, I removed some errors when trying to remove files that don't exist.

#### How I did it
create new function and add new calls for commands to get more information.
Regarding the error messages - I checked before removing the files and remove with force parameter.
 
#### How to verify it
verified that the new files and information are part of the dump

### CODE REVIEW

**Comment by volodymyrsamotiy** [comment]
@LimorAmit, could you please update PR title according to contribution guide described in README?
It should be aligned with the following format: ```[component/folder touched]: Description intent of your changes```.

File: scripts/generate_dump [hunk 1]
```diff
@@ -1523,9 +1584,11 @@ remove_secret_from_etc_files() {
     sed -i -E 's/^([^#]\S*\s*)(\S*)/\1****/g' $dumppath/etc/pam_radius_auth.conf
 
     # Remove radius passkey from per-server conf file /etc/pam_radius_auth.d/{ip}_{port}.conf
-    for filename in $dumppath/etc/pam_radius_auth.d/*.conf; do
-        sed -i -E 's/^([^#]\S*\s*)(\S*)/\1****/g' $filename
-    done
+    if [ -f "$dumppath/etc/pam_radius_auth.d/*.conf" ]; then
```
Comment (qiluo-msft) [inline scripts/generate_dump:1587]: >-f [](http://example.com/codeflow?start=9&length=2)

Why checking a wildcard filename? Is it working?

File: scripts/generate_dump [hunk 2]
```diff
@@ -700,6 +701,58 @@ save_proc() {
...
+
+    if $NOOP; then
+        echo "$CP $V -r $stats_file $TARDIR/proc_stats"
+    else
+        ( $CP $V -r $stats_file $TARDIR/proc_stats ) || echo "$stats_file error" > $TARDIR/$stats_file
+    fi
+
+    $TAR $V -rhf $TARFILE -C $DUMPDIR --mode=+rw $BASE/proc_stats
+    $RM $V -rf $TARDIR/proc_stats
+    $RM -rf $stats_file}
```
Comment (qiluo-msft) [inline scripts/generate_dump:754]: >} [](http://example.com/codeflow?start=23&length=1)

Please follow code style in the same file.

File: scripts/generate_dump [hunk 3]
```diff
@@ -700,6 +701,59 @@ save_proc() {
...
+            cat $pid/io; echo
+        else
+            #Dump also internal kernel processes if they perform writes
+            write_bytes=$(cat $pid/io | grep -w write_bytes: | cut -b 14-)
+            if [ "$write_bytes" != "0" ]; then
+                echo pid $pid - `cat $pid/comm`:
+                cat $pid/io; echo
+            fi
+        fi
+    done > $stats_file 2>&1
```
Comment (qiluo-msft) [inline scripts/generate_dump:744]: >> [](http://example.com/codeflow?start=9&length=1)

Do you want to append `>>`?

Comment (LimorAmit) [reply scripts/generate_dump]: wanted to append, fixed

**Review by LimorAmit — COMMENTED** [review]
fixed comment

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**